usersync/syncersbuilder.go

```diff
@@ -98,28 +101,88 @@ func chooseSyncerConfig(biddersSyncerConfig []namedSyncerConfig) (namedSyncerCon
 		return biddersSyncerConfig[0], nil
 	}
 
-	var bidderNames []string
-	var bidderNamesWithEndpoints []string
-	var syncerConfig namedSyncerConfig
+	var (
+		bidderNames         []string
+		nonAliasBidderNames []string
+		aliasBidderNames    []string
+		syncerConfig        namedSyncerConfig
+		parentBidderNameMap = make(map[string]struct{})
+	)
+
 	for _, bidder := range biddersSyncerConfig {
 		bidderNames = append(bidderNames, bidder.name)
 		if bidder.cfg.IFrame != nil || bidder.cfg.Redirect != nil {
-			bidderNamesWithEndpoints = append(bidderNamesWithEndpoints, bidder.name)
+			if len(bidder.bidderInfo.AliasOf) > 0 {
+				aliasBidderNames = append(aliasBidderNames, bidder.name)
+				parentBidderNameMap[bidder.bidderInfo.AliasOf] = struct{}{}
+			} else {
+				nonAliasBidderNames = append(nonAliasBidderNames, bidder.name)
+			}
 			syncerConfig = bidder
 		}
 	}
 
-	if len(bidderNamesWithEndpoints) == 0 {
+	// check if bidders have same syncer key but no endpoints
+	if len(nonAliasBidderNames)+len(aliasBidderNames) == 0 {
 		sort.Strings(bidderNames)
 		bidders := strings.Join(bidderNames, ", ")
 		return namedSyncerConfig{}, fmt.Errorf("bidders %s share the same syncer key, but none define endpoints (iframe and/or redirect)", bidders)
 	}
 
-	if len(bidderNamesWithEndpoints) > 1 {
-		sort.Strings(bidderNamesWithEndpoints)
-		bidders := strings.Join(bidderNamesWithEndpoints, ", ")
+	// check if non-alias bidders have same syncer key
+	if len(nonAliasBidderNames) > 1 {
+		sort.Strings(nonAliasBidderNames)
+		bidders := strings.Join(nonAliasBidderNames, ", ")
```

Comment:
Can this line be moved into the return statement like you did for the other returns with a similar pattern? Like how did on lines 143 and 149?